mobile_app_connector/models/account_invoice.py

```diff
@@ -146,8 +146,13 @@ def action_invoice_paid(self):
             has_app = self.env['firebase.registration'].search_count([
                 ('partner_id', '=', partner.id)
             ])
+            previous_notification = self.env["firebase.notification"].search([
+                '&', ("res_model", "=", self._name),
+                ("res_id", "=", self._uid)
```

Comment:
- `&` is the default join operator.
- res_id should be the record id and not the current user id. 

```suggestion
                ("res_model", "=", self._name),
                ("res_id", "=", invoice.id)
```

Reply:
Instead of putting the logic here, you could move it inside the `create` method of `firebase.notification` so that the behaviour is global : whenever a notification is created, if one already existed attached to a same object in odoo, it won't create it.